input, PhoneInput, live international formating of phone numbers

thomas-cleany opened the pull request:
in order to search by phone numbers, we create that component which
formats a phone number as the user is typing it. But copy/paste works
as well.
A function write the french indicator if the number starts by a 0.
Without it, the number is render as : 01 23 45 67 89.

thomas-cleany committed:
input, PhoneInput, live international formating of phone numbers

in order to search by phone numbers, we create that component which
formats a phone number as the user is typing it. But copy/paste works
as well.
A function write the french indicator if the number starts by a 0.
Without it, the number is render as : 01 23 45 67 89.

thomas-cleany commented:
There may be a problem with the configuration of the function parseInput because I've hardcoded it for french use. I could have use the a params "local" to set the indicator. But some countries like Italia keep the first number in national and international formatting.

sebastien-cleany commented on src/mui/input/PhoneInput.js:
```diff
@@ -0,0 +1,42 @@
+    if (number.charAt(0) === '0') {
+        const rest = number.slice(1);
+        number = `+33${rest}`;
+    }
```
IMO, this is handled by libphone number

sebastien-cleany replied:
```js
 var a = require('libphonenumber-js');
 var p = new a.asYouType('FR');
 p('0623576067');
```

sebastien-cleany replied:
this is working

sebastien-cleany replied:
Damn it the lib doesn't handle intl output

sebastien-cleany replied:
I do understand your hack.

sebastien-cleany replied:
Ok, first there is a bug in libphonenumber:
* in 'as you type.js' line 403 the country_code is set to undefined because metadata are array an not object.
Once the fix will be done, the nice way to handle what we need would be to do either:
* Define a custom class and overwrite full_phone_number behavior
* Do a PR to handle an output format in asYouType constructor and thus adapt full_phone_number

sebastien-cleany replied:
I just push the PR.

sebastien-cleany replied:
https://github.com/catamphetamine/libphonenumber-js/pull/149

sebastien-cleany replied:
Already merged and published by the maintener

thomas-cleany committed:
Merge branch 'local' of github.com:cleany/admin-on-rest into feature_phone_input

thomas-cleany committed:
PhoneInput: handling national numbers to international

when we write a national number in the locale defined he's rendered
as national number. For example : 0123456789 => 01 23 45 67 89
So we extends asYouType to render all the numbers in international
format.
Bilan : 0123456789 => +33 1 23 45 67 89

thomas-cleany committed:
PhoneInput: return input if the number can't be formatted
